frontend/app/src/next/components/runs/trigger-run-modal.tsx

```diff
@@ -258,218 +289,233 @@ function TriggerRunModalContent({
         </DialogHeader>
 
         <div className="space-y-4">
-          {selectedRunId && (
+          {selectedRunId && !disabledCapabilities.includes('fromRecent') && (
             <RunDetailProvider runId={selectedRunId}>
               <WithPreviousInput
                 setInput={setInput}
                 setAddlMeta={setAddlMeta}
               />
             </RunDetailProvider>
           )}
-          <div>
-            <label className="text-sm font-medium">Workflow</label>
-            <select
-              className="w-full mt-1 rounded-md border border-input bg-background px-3 py-2"
-              value={selectedWorkflowId}
-              onChange={(e) => {
-                setSelectedWorkflowId(e.target.value);
-                setSelectedRunId(''); // Reset selected run when workflow changes
-              }}
-            >
-              <option value="">Select a workflow</option>
-              {workflows?.map((workflow: Workflow) => (
-                <option key={workflow.metadata.id} value={workflow.metadata.id}>
-                  {workflow.name}
-                </option>
-              ))}
-            </select>
-          </div>
-
-          <div>
-            <label className="text-sm font-medium flex items-center gap-2">
-              <FaCodeBranch className="text-muted-foreground" size={16} />
-              From Recent Run
-            </label>
-            <select
-              className="w-full mt-1 rounded-md border border-input bg-background px-3 py-2 disabled:opacity-50 disabled:cursor-not-allowed"
-              value={selectedRunId}
-              disabled={!selectedWorkflowId}
-              onChange={(e) => {
-                const runId = e.target.value;
-                setSelectedRunId(runId);
-              }}
-            >
-              <option value="">Select a recent run</option>
-              {recentRuns
-                ?.filter((run) => run.workflowId === selectedWorkflowId)
-                .map((run) => (
-                  <option key={run.metadata.id} value={run.metadata.id}>
-                    {getFriendlyWorkflowRunId(run)}
+          {!disabledCapabilities.includes('workflow') && (
+            <div>
+              <label className="text-sm font-medium">Workflow</label>
+              <select
+                className="w-full mt-1 rounded-md border border-input bg-background px-3 py-2"
+                value={selectedWorkflowId}
+                onChange={(e) => {
+                  setSelectedWorkflowId(e.target.value);
+                  setSelectedRunId(''); // Reset selected run when workflow changes
```

Comment:
nit but something I really dislike about the AI-gen code 😅 I really dislike comments like this - it's obvious if you're reading the code what's going on here, and if it's not that's a problem with how the code is written IMO

Reply:
yeah, i agree